src/main/java/seedu/ccacommander/logic/commands/RedoCommand.java

```diff
@@ -0,0 +1,31 @@
+package seedu.ccacommander.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.ccacommander.model.Model.PREDICATE_SHOW_ALL_EVENTS;
+import static seedu.ccacommander.model.Model.PREDICATE_SHOW_ALL_MEMBERS;
+
+import seedu.ccacommander.logic.commands.exceptions.CommandException;
+import seedu.ccacommander.model.Model;
+
+/**
+ * Redoes the latest redoable command {@code Command}
+ */
+public class RedoCommand extends Command {
+    public static final String COMMAND_WORD = "redo";
+    public static final String MESSAGE_SUCCESS_REDO = "Command successfully redone: \n%s";
+    public static final String MESSAGE_NO_AVAILABLE_COMMAND = "No available commands to redo.";
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        if (!model.canRedo()) {
+            throw new CommandException(MESSAGE_NO_AVAILABLE_COMMAND);
+        }
+
+        String undoCommandMessage = model.redo();
+        model.updateFilteredMemberList(PREDICATE_SHOW_ALL_MEMBERS);
+        model.updateFilteredEventList(PREDICATE_SHOW_ALL_EVENTS);
+        return new CommandResult(String.format(MESSAGE_SUCCESS_REDO, undoCommandMessage));
```

Comment:
Should this be redoCommandMessage instead?
```suggestion
        String redoCommandMessage = model.redo();
        model.updateFilteredMemberList(PREDICATE_SHOW_ALL_MEMBERS);
        model.updateFilteredEventList(PREDICATE_SHOW_ALL_EVENTS);
        return new CommandResult(String.format(MESSAGE_SUCCESS_REDO, redoCommandMessage));
```

Reply:
Agree with this, will make the change!